File: python_tests/vips_shrink.py
#!/usr/bin/python

import sys
from timer import TimerContext, printResults

# Add the path to vipsCC by hand so I don't have to mess around with environment variables.
sys.path += ['/usr/local/lib/python2.7/site-packages/']

from vipsCC import *


def main(imgfile, percentages):
    with TimerContext("# " + imgfile, "Loading"):
        im = VImage.VImage(imgfile, "r")

    for pc in percentages:
        if pc <= 0 or pc > 100:
            print "Illegal percentage: ", pc
            exit(1)

        outfile = "%d-vips-py-%s" % (pc, imgfile)
        sz = round(100.0/pc)
        with TimerContext(imgfile, str(pc)):
            im.shrink(sz, sz).write(outfile)

    printResults()
                                


if len(sys.argv) < 3:
    print "Usage: shrink.py <filename> <percentage> ..."
    exit(1)

fname = sys.argv[1]
percentages = map(lambda(x): int(x), sys.argv[2:])
main(fname, percentages)

File: ruby_tests/vips_shrink.rb
#!/usr/bin/env ruby


require 'rubygems'
require 'vips'

# Add path to local modules
proc {
  $LOAD_PATH.unshift(File.dirname(__FILE__))
}.call()

require 'timer'

include VIPS
include Timer

def main(imgfile, percentages)
  im = nil      # declare in the local scope
  time(imgfile, "Loading") {im = Image.new(imgfile)}

  base = File.basename(imgfile)

  percentages.each do |p|
    percent = p.to_f
    raise "Invalid percentage: #{p}" unless percent > 0 && percent <= 100
    sz = (100.0/percent).round
    time imgfile, percent.to_i.to_s do
      im.shrink(sz).write("#{percent.round}-vips-rb-#{base}")
    end
  end

  print_timings()
end

if ARGV.size < 2
  puts "USAGE: shrink <filename> <percentage> ..."
  exit 1
end

main ARGV[0], ARGV[1..-1]
